src/olympia/editors/views.py

```diff
@@ -534,18 +534,56 @@ def application_versions_json(request):
     return {'choices': f.version_choices_for_app_id(app_id)}
 
 
+def _get_deleted_versions(addon):
+    class PseudoVersion(object):
+        def __init__(self):
+            self.all_activity = []
+
+        all_files = ()
+        approvalnotes = None
+        compatible_apps_ordered = ()
+        releasenotes = None
+        status = 'Deleted'
+        deleted = True
+        channel = amo.RELEASE_CHANNEL_LISTED
+
+        @property
+        def created(self):
+            return self.all_activity[0].created
+
+        @property
+        def version(self):
+            return (self.all_activity[0].activity_log
+                        .details.get('version', '[deleted]'))
+
+    # Grab review history for deleted versions of this add-on
+    # Version are soft-deleted now but we need this for older deletions.
+    comments = (CommentLog.objects
+                .filter(activity_log__action__in=amo.LOG_REVIEW_QUEUE,
+                        activity_log__versionlog=None,
+                        activity_log__addonlog__addon=addon)
+                .order_by('created')
+                .select_related('activity_log'))
+
+    comment_versions = defaultdict(PseudoVersion)
+    for c in comments:
+        c.version = c.activity_log.details.get('version', c.created)
+        comment_versions[c.version].all_activity.append(c)
+    return comment_versions.values()
+
+
 @addons_reviewer_required
 @addon_view_factory(qs=Addon.unfiltered.all)
-def review(request, addon):
-    # This is a short term fix that will prevent all but admin reviewers from
-    # looking at addons with mixed unlisted/listed versions.
-    if (addon.has_unlisted_versions() and
-            not acl.check_unlisted_addons_reviewer(request)):
+def review(request, addon, channel=None):
+    # channel is passed in as text, but we want the constant.
+    channel = amo.CHANNEL_CHOICES_LOOKUP.get(
+        channel, amo.RELEASE_CHANNEL_LISTED)
+    unlisted_only = (channel == amo.RELEASE_CHANNEL_UNLISTED or
+                     not addon.has_listed_versions())
+    if unlisted_only and not acl.check_unlisted_addons_reviewer(request):
         raise http.Http404
```

Comment:
Not sure - a 403 would expose the add-ons existence vs. an invalid slug (which would result in a 404 in the @addon_view).  But can do.